src/ess_rhat.jl

```diff
@@ -233,25 +236,206 @@ The ESS and ``\\widehat{R}`` values can be computed for the following estimators
 - `StatsBase.mad`
 - `Base.Fix2(Statistics.quantile, p::Real)`
 
+## Types
+
+If no `estimator` is provided, the following `type`s of ESS estimates may be computed:
+- `:bulk`/`:rank`: mean-ESS computed on rank-normalized draws. This type diagnoses poor
+    convergence in the bulk of the distribution due to trends or different locations of the
+    chains.
+- `:tail`: minimum of the quantile-ESS for the symmetric quantiles where
+    `tail_prob=0.1` is the probability in the tails. This type diagnoses poor convergence in
+    the tails of the distribution. If this type is chosen, `kwargs` may contain a
+    `tail_prob` keyword.
+- `:basic`: basic ESS, equivalent to specifying `estimator=Statistics.mean`.
+
+While Bulk-ESS is conceptually related to basic ESS, it is well-defined even if the chains
+do not have finite variance.[^VehtariGelman2021]. For each parameter, rank-normalization
+proceeds by first ranking the inputs using "tied ranking" and then transforming the ranks to
+normal quantiles so that the result is standard normally distributed. This transform is
+monotonic.
+
 [^VehtariGelman2021]: Vehtari, A., Gelman, A., Simpson, D., Carpenter, B., & Bürkner, P. C. (2021).
     Rank-normalization, folding, and localization: An improved ``\\widehat {R}`` for
     assessing convergence of MCMC. Bayesian Analysis.
     doi: [10.1214/20-BA1221](https://doi.org/10.1214/20-BA1221)
     arXiv: [1903.08008](https://arxiv.org/abs/1903.08008)
 """
-function ess_rhat(samples::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
-    return ess_rhat(Statistics.mean, samples; kwargs...)
+@constprop :aggressive function ess(
+    samples::AbstractArray{<:Union{Missing,Real},3};
+    estimator=nothing,
+    type=nothing,
```

Comment:
Maybe restrict the types of the keyword arguments here since we know the admissible types? Also, maybe it would be simpler to use `:basic` as the default value here?
```suggestion
    type::Symbol = :basic,
```
I'm not a fan of supporting both `Val` and `Symbol` but in that case we could use
```suggestion
    type::Union{Val,Symbol}=Val(:basic),
```